web/packages/teleport/src/Console/DocumentKubeExec/DocumentKubeExec.story.tsx

```diff
@@ -0,0 +1,136 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+
+import { createTeleportContext } from 'teleport/mocks/contexts';
+import ConsoleCtx from 'teleport/Console/consoleContext';
+
+import { ContextProvider } from 'teleport';
+import { TestLayout } from 'teleport/Console/Console.story';
+import TeleportContext from 'teleport/teleportContext';
+import * as stores from 'teleport/Console/stores/types';
+
+import DocumentKubeExec from './DocumentKubeExec';
+
+import type { Session } from 'teleport/services/session';
+
+export default {
+  title: 'Teleport/Console/DocumentKubeExec',
+};
+
+export const Connected = () => {
+  const { ctx, consoleCtx } = getContexts();
+
+  return (
+    <DocumentKubeExecWrapper ctx={ctx} consoleCtx={consoleCtx} doc={baseDoc} />
+  );
+};
+
+export const NotFound = () => {
+  const { ctx, consoleCtx } = getContexts();
+
+  const disconnectedDoc = {
+    ...baseDoc,
+    status: 'disconnected' as const,
+  };
+
+  return (
+    <DocumentKubeExecWrapper
+      ctx={ctx}
+      consoleCtx={consoleCtx}
+      doc={disconnectedDoc}
+    />
+  );
+};
+
+export const ServerError = () => {
+  const { ctx, consoleCtx } = getContexts();
+
+  const noSidDoc = {
+    ...baseDoc,
+    sid: '',
+  };
+
+  return (
+    <DocumentKubeExecWrapper ctx={ctx} consoleCtx={consoleCtx} doc={noSidDoc} />
+  );
+};
+
+export type Props = {
+  ctx: TeleportContext;
+  consoleCtx: ConsoleCtx;
+  doc: stores.DocumentKubeExec;
+};
+
+export const DocumentKubeExecWrapper = ({ ctx, consoleCtx, doc }) => {
+  return (
+    <ContextProvider ctx={ctx}>
+      <TestLayout ctx={consoleCtx}>
+        <DocumentKubeExec doc={doc} visible={true} />
+      </TestLayout>
+    </ContextProvider>
+  );
+};
+
+export function getContexts() {
+  const ctx = createTeleportContext();
+  const consoleCtx = new ConsoleCtx();
+  const tty = consoleCtx.createTty(session);
+  tty.connect = () => null;
+  consoleCtx.createTty = () => tty;
+  consoleCtx.storeUser = ctx.storeUser;
+
+  return { ctx, consoleCtx };
+}
+
+export const baseDoc = {
+  kind: 'kubeExec',
+  status: 'connected',
+  sid: 'sid-value',
+  clusterId: 'clusterId-value',
+  serverId: 'serverId-value',
+  login: 'login-value',
+  kubeCluster: 'kubeCluster1',
+  kubeNamespace: 'namespace1',
+  pod: 'pod1',
+  container: '',
+  id: 3,
+  url: 'fd',
+  created: new Date(),
+  command: '/bin/bash',
+  isInteractive: true,
+} as const;
```

Comment:
That's one of the problems with exporting stuff out of stories and then importing it in tests. On a conceputal level, I think it's not what one's supposed to (though that's the pattern I think we've been following in a few places).

But a practical problem with it is that Storybook expects each exported member to be a React component, so we end up with a bunch of items that look like stories, but they just don't work.

<img width="900" alt="exported-stories" src="https://github.com/gravitational/teleport/assets/27113/76d63177-e7de-49ed-96c4-4782f9337318">